app/assets/javascripts/govuk_publishing_components/components/layout-super-navigation-header.js

```diff
@@ -126,7 +126,56 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     toggle($target, $targetMenu)
   }
 
+  SuperNavigationMegaMenu.prototype.handleKeyDown = function (event) {
+    var KEY_TAB = 9
+    var KEY_ESC = 27
+    var $navToggle = this.$module.querySelector('.gem-c-layout-super-navigation-header__navigation-top-toggle-button')
+    var $navMenu = this.$module.querySelector('.gem-c-layout-super-navigation-header__navigation-dropdown-menu')
+    var $menuLinks = $navMenu.querySelectorAll('li')
+
+    if (event.keyCode === KEY_TAB) {
+      if (!$navMenu.hasAttribute('hidden')) {
+        switch (document.activeElement) {
+          case $navToggle:
+            if (!event.shiftKey) {
+              event.preventDefault()
+              $menuLinks[0].querySelector('a').focus()
+            }
+            break
+          case $menuLinks[$menuLinks.length - 1].querySelector('a'):
+            if (!event.shiftKey) {
+              event.preventDefault()
+              this.$searchToggle.focus()
+            }
+            break
+          case $menuLinks[0].querySelector('a'):
+            if (event.shiftKey) {
+              event.preventDefault()
+              $navToggle.focus()
+            }
+            break
+          case this.$searchToggle:
+            if (event.shiftKey) {
+              event.preventDefault()
+              $menuLinks[$menuLinks.length - 1].querySelector('a').focus()
+            }
+            break
+          default:
+            break
+        }
+      }
+    } else if (event.keyCode === KEY_ESC) {
+      if (!$navMenu.hasAttribute('hidden')) {
+        hide($navToggle, $navMenu)
+        this.$searchToggle.focus()
+      }
+    }
```

Comment:
I just want to double-check what behaviour should be here when the open menu is closed using the escape key.

At the moment it does match what is in the Trello card, in that, when you press the Esc key to close the menu, focus jumps to the search menu toggle button, but not sure where that recommendation has come from, I cannot see this anywhere in the comments or from the DAC audit.

Looking at the WCAG documentation, I think the focus state should jump back to the Menu toggle button instead - https://www.w3.org/WAI/WCAG20/Techniques/failures/F85, but would be good to check with an accessible specialist on this.

Lastly, do we also need to add the escape key functionality to the search menu as well?

Reply:
Good point on the expected behaviour on pressing the escape key when the Navigation Menu is open. Wider opinion agrees with you that this should return the user to the means by which they opened it - i.e. the Menu toggle button so this has been updated here.

Although it isn't part of the audit report it would seem to make sense to apply the same functionality to the Search Menu so that has been done as well. 

Additionally the scenario of tabbing from the final link in the navigation has also been added to the Search Menu since this will otherwise leave that menu open as the user tabs through the rest of the page.

Reply:
Nice!, thanks for double-checking and adding the additional functionality to the Search Menu as well, works great!